lib/bolt/node/ssh.rb

```diff
@@ -24,7 +24,7 @@ def connect
         non_interactive: true
       }
 
-      options[:port] = @port if @port
+      options[:port] = port if port
```

Comment:
regarding dropping the port/host/uri methods on Nod: this would `options[:port] = target.port if target.port`